### PR TITLE
Unify zsh and bash handling, and add a test

### PR DESCRIPTION
**Pull request by burke**
Now that we've whittled away more of the cases where DEBUG is run, it actually looks a lot like zsh's `preexec_functions`, so let's just masquerade as that.

Drop the zsh `chpwd_functions` registration. This is completely redundant with the hook firing for both `preexec` and `precmd`.

Also drop support for fish. The README documents why.

The test is pretty disgusting but does a bit of basic sanity checking.

### CODE REVIEW

**Review by donk-shopify — APPROVED** [review]
LGTM

File: test [hunk 1]
```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/ruby --disable-gems
+
+# We're being very lazy here and assuming macOS with some preconditions:
```
Comment (lavoiesl) [inline test:3]: I would say it's worth setting a basic rspec / minitest / mocha and hooking up travis for linux and macos testing: https://docs.travis-ci.com/user/reference/osx/

**Review by lavoiesl — APPROVED** [review]
I like the simplification 👍

File: test (hunk 1 above)
Comment (burke) [reply test:3]: Yeah, I will probably do that next time I have to interact with this. I ran out of quality-improvement-motivation for the afternoon.

**Pull request merged**